showcase/src/main/kotlin/com/recco/showcase/ShowcaseRepository.kt

```diff
@@ -0,0 +1,82 @@
+package com.recco.showcase
+
+import android.content.Context
+import com.recco.api.model.ReccoFont
+import com.recco.api.model.ReccoPalette
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+class ShowcaseRepository @Inject constructor(
+    @ApplicationContext context: Context
+) {
+    private val prefs by lazy {
+        context.getSharedPreferences("showcase", Context.MODE_PRIVATE)
+    }
+
+    fun setUserId(userId: String) {
+        with(prefs.edit()) {
+            putString(USER_ID_KEY, userId)
+            apply()
+        }
+    }
+
+    fun clearUserId() {
+        with(prefs.edit()) {
+            putString(USER_ID_KEY, null)
+            apply()
+        }
+    }
+
+    fun isUserLoggedIn() = !prefs.getString(USER_ID_KEY, "").isNullOrBlank()
+
+    fun setReccoPalette(palette: ReccoPalette) {
+        with(prefs.edit()) {
+            putString(RECCO_PALETTE_KEY, palette.asName())
+            apply()
+        }
+    }
+
+    fun getSelectedReccoPalette(): ReccoPalette {
+        val reccoPaletteName = prefs.getString(RECCO_PALETTE_KEY, "")
+        if (reccoPaletteName.isNullOrBlank()) return ReccoPalette.Fresh
+        return reccoPaletteName.asPalette()
```

Comment:
wdyt about this approach?, for me is easier to read:
```kotlin
return if (reccoPaletteName.isNullOrBlank()) {
    ReccoPalette.Fresh
}  else {
    reccoPaletteName.asPalette()
}
```

Reply:
Done 👍